TestApp/Repositories/DataAccessRepositories.cs

```diff
@@ -0,0 +1,285 @@
+        bool DeleteIdentifierToPerson(Guid pid, Guid IdenId);
+        bool UpdatePersonWithoutIdentifier(Person person);
+        bool DeletePersonVirtually(Guid id);
+        //TEntity GetBook(U id);
+        //int AddBook(TEntity b);
+        //int UpdateBook(U id, TEntity b);
+        //int DeleteBook(U id);
+    }
+
+    public class DataAccessRepository : IDataAccess<Person, int>
+    {
+        TestAPIContext ctx;
+        public DataAccessRepository(TestAPIContext c)
+        {
+            ctx = c;
+        }
+        //public int AddBook(Book b)
+        //{
+        //    ctx.Books.Add(b);
+        //    int res = ctx.SaveChanges();
+        //    return res;
+        //}
+
+        //public int DeleteBook(int id)
+        //{
+        //    int res = 0;
+        //    var book = ctx.Books.FirstOrDefault(b => b.Id == id);
+        //    if (book != null)
+        //    {
+        //        ctx.Books.Remove(book);
+        //        res = ctx.SaveChanges();
+        //    }
+        //    return res;
+        //}
+
+        //public Book GetBook(int id)
+        //{
+        //    var book = ctx.Books.FirstOrDefault(b => b.Id == id);
+        //    return book;
+        //}
+
+        public IEnumerable<Person> GetPersonList()
+        {
+            try
+            {
+                var persons = ctx.Persons.Where(x=>x.isDeleted==false).ToList();
+                return persons;
+            }
+            catch (Exception ex)
+            {
+                return null;
+            }
+            
+        }
+        public IEnumerable<Person> GetPersonListByFirstOrLastName(string FName, string LName)
+        {
+            try
+            {
+                if (FName.Trim().ToString() == "" && FName.Trim().ToString() == "")
+                {
+                    var persons = ctx.Persons.Where(x=>x.isDeleted==false).ToList();
+                    return persons;
+                } else if(FName.Trim().ToString() == "")
+                {
+                    var persons = ctx.Persons.Where(x=>x.LastName==LName && x.isDeleted==false).ToList();
+                    return persons;
+                } else
+                {
+                    var persons = ctx.Persons.Where(x => x.FirstName== FName && x.isDeleted==false).ToList();
+                    return persons;
+                }
+
+            }
+            catch(Exception ex)
+            {
+                return null;
+            }
+        }
+        public IEnumerable<Person> GetPersonListBySpecificIdentity(string specification)
+        {
+            try
+            {
+                if (specification.Trim().ToString() == "")
+                {
+                    var persons = ctx.Persons.ToList();
+                    return persons;
+                }                
+                else
+                {
+                    var number = (int)((IdentificationType)Enum.Parse(typeof(IdentificationType), specification));
```

Comment:
this is really wrong, look for a different way. We cannot support that for each identification found we need to create another query to go ahead and all the identities. Really this is N + 1 problem

Reply:
Can you please suggest other.

Reply:
You can use include or custom query